dandi/download.py

```diff
@@ -124,19 +145,40 @@ def download(
         authenticate=authenticate,
         progressbars=True,  # TODO: redo all this
     )
+
+    lgr.info(f"Downloading {asset_type} with id {asset_id} from {girder_server_url}")
+
+    # there might be multiple asset_ids, e.g. if multiple files were selected etc,
+    # so we will traverse all of them
+    files = flattened(
+        _get_asset_files(
+            asset_id_, asset_type, output_dir, client, authenticate, existing, recursive
+        )
+        for asset_id_ in set(flattened([asset_id]))
+    )
+
+    for file in files:
+        client.download_file(
+            file["id"],
+            op.join(output_dir, file["path"]),
+            existing=existing,
+            attrs=file["attrs"],
+        )
+
+
+def _get_asset_files(
+    asset_id, asset_type, output_dir, client, authenticate, existing, recursive
+):
     # asset_rec = client.getResource(asset_type, asset_id)
     # lgr.info("Working with asset %s", str(asset_rec))
-
     # In principle Girder's client already has ability to download any
     # resource (collection/folder/item/file).  But it seems that "mending" it
     # with custom handling (e.g. later adding filtering to skip some files,
     # or add our own behavior on what to do when files exist locally, etc) would
     # not be easy.  So we will reimplement as a two step (kinda) procedure.
-
```

Comment:
I believe I didn't do any of those -- must be "pre-commit" or some other evil automation